Linking against SUNDIALS 7.

#16879 helped compile deal.II with SUNDIALS 7. Unfortunately, this was not enough to ensure compatibility.

Linking executables of user applications, tutorial examples or even tests against deal.II then fails with the following undefined references.
```
[...]/dealii-9.6.0-pre/lib/libdeal_II.g.so.9.6.0-pre: error: undefined reference to 'SUNContext_Create'
[...]/dealii-9.6.0-pre/lib/libdeal_II.g.so.9.6.0-pre: error: undefined reference to 'SUNContext_Free'
[...]/dealii-9.6.0-pre/lib/libdeal_II.g.so.9.6.0-pre: error: undefined reference to 'SUNLinSolNewEmpty'
[...]/dealii-9.6.0-pre/lib/libdeal_II.g.so.9.6.0-pre: error: undefined reference to 'SUNMatNewEmpty'
[...]/dealii-9.6.0-pre/lib/libdeal_II.g.so.9.6.0-pre: error: undefined reference to 'SUNMatDestroy'
[...]/dealii-9.6.0-pre/lib/libdeal_II.g.so.9.6.0-pre: error: undefined reference to 'SUNLinSolFree'
[...]/dealii-9.6.0-pre/lib/libdeal_II.g.so.9.6.0-pre: error: undefined reference to 'N_VNewEmpty'
[...]/dealii-9.6.0-pre/lib/libdeal_II.g.so.9.6.0-pre: error: undefined reference to 'N_VCopyOps'
[...]/dealii-9.6.0-pre/lib/libdeal_II.g.so.9.6.0-pre: error: undefined reference to 'N_VDestroy'
[...]/dealii-9.6.0-pre/lib/libdeal_II.g.so.9.6.0-pre: error: undefined reference to 'N_VFreeEmpty'
[...]/dealii-9.6.0-pre/lib/libdeal_II.g.so.9.6.0-pre: error: undefined reference to 'SUNLinSolFreeEmpty'
collect2: error: ld returned 1 exit status
```

In the linking command in my user code I spot
```
[...]/sundials-7.0.0/lib
[...]/sundials-7.0.0/lib/libsundials_idas.so 
[...]/sundials-7.0.0/lib/libsundials_arkode.so
[...]/sundials-7.0.0/lib/libsundials_kinsol.so
[...]/sundials-7.0.0/lib/libsundials_nvecserial.so 
[...]/sundials-7.0.0/lib/libsundials_nvecparallel.so
```

The sundials folder contains more object files than that. Maybe that's the issue?
```
$ find [...]/sundials-7.0.0/lib -name "*.so"
./libsundials_arkode.so
./libsundials_nvecparallel.so
./libsundials_ida.so
./libsundials_sunnonlinsolfixedpoint.so
./libsundials_sunlinsolspfgmr.so
./libsundials_sunnonlinsolnewton.so
./libsundials_kinsol.so
./libsundials_cvode.so
./libsundials_nvecserial.so
./libsundials_sunlinsolpcg.so
./libsundials_sunmatrixdense.so
./libsundials_sunlinsolsptfqmr.so
./libsundials_sunlinsolspbcgs.so
./libsundials_sunmatrixband.so
./libsundials_sunlinsolspgmr.so
./libsundials_sunmatrixsparse.so
./libsundials_cvodes.so
./libsundials_core.so
./libsundials_sunlinsolband.so
./libsundials_sunlinsoldense.so
./libsundials_nvecmanyvector.so
./libsundials_idas.so
./libsundials_nvecmpiplusx.so
./libsundials_nvecmpimanyvector.so
```

Do you have an idea how to get around this? I didn't find anything related to linking in the changelog.

--- a/cmake/modules/FindDEAL_II_SUNDIALS.cmake
+++ b/cmake/modules/FindDEAL_II_SUNDIALS.cmake
@@ -38,6 +38,11 @@
 set(SUNDIALS_DIR "" CACHE PATH "An optional hint to a SUNDIALS_DIR installation")
 set_if_empty(SUNDIALS_DIR "$ENV{SUNDIALS_DIR}")
 
+deal_ii_find_library(SUNDIALS_LIB_CORE NAMES sundials_core
+  HINTS ${SUNDIALS_DIR}
+  PATH_SUFFIXES lib${LIB_SUFFIX} lib64 lib
+  )
+
 deal_ii_find_library(SUNDIALS_LIB_IDAS NAMES sundials_idas
   HINTS ${SUNDIALS_DIR}
   PATH_SUFFIXES lib${LIB_SUFFIX} lib64 lib
@@ -121,8 +126,17 @@ if(NOT SUNDIALS_CONFIG_H MATCHES "-NOTFOUND")
     )
 endif()
 
+#
+# sundials_core is only required as of version 7.0.0.
+#
+set(_sundials_lib_core)
+if(SUNDIALS_VERSION VERSION_GREATER_EQUAL 7)
+  set(_sundials_lib_core "SUNDIALS_LIB_CORE")
+endif()
+
 process_feature(SUNDIALS
   LIBRARIES REQUIRED
+    ${_sundials_lib_core}
     ${_sundials_lib_ida}
     SUNDIALS_LIB_ARKODE
     SUNDIALS_LIB_KINSOL
@@ -131,6 +145,7 @@ process_feature(SUNDIALS
   INCLUDE_DIRS REQUIRED
     SUNDIALS_INCLUDE_DIR
   CLEAR
+    SUNDIALS_LIB_CORE
     SUNDIALS_LIB_IDA
     SUNDIALS_LIB_IDAS
     SUNDIALS_LIB_ARKODE

